src/jit/codegencommon.cpp

```diff
@@ -9450,6 +9450,13 @@ void CodeGen::genFnEpilog(BasicBlock* block)
     genRestoreCalleeSavedFltRegs(compiler->compLclFrameSize);
 #endif // !FEATURE_STACK_FP_X87
 
+#ifdef JIT32_GCENCODER
+    // On IA32, we start the OS-reported portion of the epilog after restoring any callee-saved
+    // floating-point registers. This avoids the need to update the x86 unwinder and retains binary
+    // compatibility between later versions of the JIT and earlier versions of the runtime.
+    getEmitter()->emitStartEpilog();
```

Comment:
> If we were to change the ABI convention to include callee-saved xmm/ymm registers (which perhaps we should do, for CQ/performance reasons), we'd need to update the unwinder, and move this line of code.

Yeah, this gets tricky. Avoiding the AVX -> SSE transition penalty at the end of a function requires that the `vzeroupper` instruction executes after all AVX instructions in the function, including those that may be used to restore callee-save FP registers. In other words, unless we ensured that the restores only used SSE instructions, we could not move the `vzeroupper` instruction before the restores and it would therefore end up as part of the epilog. At that point we'd need to update the unwinder to recognize `vzeroupper` which would obivate the need for this change.

I agree that the comment is not terribly clear--I'll see if I can think of a better wording.